Add language_csharp.lua

DevinCarr opened the pull request.

rxi commented:
Looks good with the exception of this part:

```lua
style.syntax["character"] = style.syntax["string"]
```

In the case of loading a new colour theme the old colour theme's string value would still be used for characters which makes doing something like this a problem. The approach used by other language syntaxes in lite is to make the character literals resolve to the string token directly. Happy to merge if you don't mind making this change!

DevinCarr commented:
Makes perfect sense to me, I have removed that change and targeted "string" for the syntax highlighting.

Also, I updated a few things I forgot and realized this afternoon. Should be good to merge now!

rxi commented:
Looks good! Thanks
